x-pack/plugins/ml/common/util/string_utils.ts

```diff
@@ -22,3 +22,20 @@ export function getMedianStringLength(strings: string[]) {
   const sortedStringLengths = strings.map((s) => s.length).sort((a, b) => a - b);
   return sortedStringLengths[Math.floor(sortedStringLengths.length / 2)] || 0;
 }
+
+/**
+ * Creates a deterministic number based hash out of a string.
+ */
+export function stringHash(str: string): number {
+  let hash = 0;
+  let chr = 0;
+  if (str.length === 0) {
+    return hash;
+  }
+  for (let i = 0; i < str.length; i++) {
+    chr = str.charCodeAt(i);
+    hash = (hash << 5) - hash + chr; // eslint-disable-line no-bitwise
+    hash |= 0; // eslint-disable-line no-bitwise
+  }
+  return hash < 0 ? hash * -2 : hash;
+}
```

Comment:
we already have this function defined in `x-pack/plugins/ml/common/util/group_color_utils.ts`

Reply:
I moved that function to this file because it's now used more generally out of the context of color utils.